src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -178,15 +190,15 @@ private void visitObject(MaskingState maskingState, @CheckForNull KeyMaskingConf
                 }
                 visitValue(maskingState, keyMaskingConfig);
             }
+            maskingState.backtrackCurrentJsonPath();
+
             stepOverWhitespaceCharacters(maskingState);
-            if (AsciiCharacter.isComma(maskingState.byteAtCurrentIndex())) {
-                maskingState.incrementCurrentIndex(); // step over comma separating elements
+            // check if we're at the end of a (non-empty) object
+            if (maskingState.byteAtCurrentIndex() == '}') {
+                break;
             }
-            stepOverWhitespaceCharacters(maskingState);
-
-            maskingState.backtrackCurrentJsonPath();
         }
-        maskingState.incrementCurrentIndex(); // step over closing curly bracket
+        maskingState.next();
```

Comment:
Nit: it is nice to add the comment back (step over closing curly bracket) but if this is the only comment I can do it in my next PR